code/datums/components/singularity.dm

```diff
@@ -225,6 +225,8 @@
 				break
 			// eat the stuff if we're going to move into it so it doesn't mess up our movement
 			for(var/atom/thing_on_turf in current_turf.contents)
+				if(thing_on_turf == parent)
+					continue
 				consume(src, thing_on_turf)
 			consume(src, current_turf)
```

Comment:
This should be impossible unless something inside singularity_act sleeps, unless consuming a turf causes everything on it to be consumed too